pkg/sql/sem/builtins/all_builtins.go

```diff
@@ -14,63 +14,88 @@ import (
 	"fmt"
 	"sort"
 
-	"github.com/cockroachdb/cockroach/pkg/server/telemetry"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/builtinsregistry"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
-	"github.com/cockroachdb/cockroach/pkg/sql/sqltelemetry"
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
 )
 
-// AllBuiltinNames is an array containing all the built-in function
+// orderedStrings sorts a slice of strings lazily
+// for better performance.
+type orderedStrings struct {
+	strings []string
+	sorted  bool
+}
+
+func (o *orderedStrings) add(s string) {
+	if o.sorted {
+		o.insert(s)
+	} else {
+		o.strings = append(o.strings, s)
+	}
+}
+
+func (o *orderedStrings) sort() {
+	if !o.sorted {
+		sort.Strings(o.strings)
+	}
+	o.sorted = true
+}
+
+func (o *orderedStrings) insert(s string) {
+	i := sort.SearchStrings(o.strings, s)
+	o.strings = append(o.strings, "")
+	copy(o.strings[i+1:], o.strings[i:])
+	o.strings[i] = s
+}
+
+var allBuiltinNames orderedStrings
+
+// AllBuiltinNames returns a slice containing all the built-in function
 // names, sorted in alphabetical order. This can be used for a
 // deterministic walk through the Builtins map.
-var AllBuiltinNames []string
+func AllBuiltinNames() []string {
```

Comment:
+1, I'm not sure why this function is preferred over a sorted string slice. Trying make the string slice private maybe? If so, I think we may just sort it once in the `init` function right?

Reply:
This actually just sorts the slice once--further calls to .sort() will short-circuit. It's happening here rather than in an init() so that we don't depend on load order to fully populate the slices before they're sorted.

Reply:
I see, thanks, for some reason I missed the sortedStrings implementation above.